packages/react-components/src/components/timegraph-output-component.tsx

```diff
@@ -707,9 +714,87 @@ export class TimegraphOutputComponent extends AbstractTreeOutputComponent<Timegr
             resolution: newResolution
         };
 
+        // Can we make this cleaner?
+        this.setNextAndPrevious(rows);
+        this.markersChartLayer.selectedMarker = undefined;
         this.setState({ markerCategoryEntries: annotationEntries, markerLayerData: markerLayerData });
     }
 
+    /**
+     * Sets next and prev for marker navigation with < & >.
+     * Directly modifies the input.
+     * @param rowsMap TimeGraphRowModel[]
+     */
+    private setNextAndPrevious = (rowsMap: TimelineChart.TimeGraphRowModel[]) => {
```

Comment:
Not sure if we should do that to avoid unnecessary processing. I put a comment regarding this in https://github.com/eclipse-cdt-cloud/timeline-chart/pull/216.